cycle.go

```diff
@@ -39,10 +39,10 @@ type errCycleDetected struct {
 func (e errCycleDetected) Error() string {
```

Comment:
nit: Consider renaming this file to cycle_error.go.
nit: Rename `cycleEntry` into something more related to the error.